x-pack/plugins/stack_connectors/server/connector_types/gemini/gemini.test.ts

```diff
@@ -0,0 +1,246 @@
+    query: {},
+    body: {},
+    httpVersion: '',
+    protocol: 'http1',
+  };
+
+  beforeEach(() => {
+    mockServiceParams = {
+      connector: { id: 'Gemini-id', type: 'gemini' },
+      configurationUtilities: mockConfigurationUtilities,
+      config: {
+        apiUrl: 'https://api.gemini.com',
+        defaultModel: 'gemini-1.5-pro-preview-0409',
+        gcpRegion: 'us-central1',
+        gcpProjectID: 'my-project-12345',
+      },
+      secrets: {
+        credentialsJson: JSON.stringify({
+          type: 'service_account',
+          project_id: '',
+          private_key_id: '',
+          private_key: '-----BEGIN PRIVATE KEY----------END PRIVATE KEY-----\n',
+          client_email: '',
+          client_id: '',
+          auth_uri: 'https://accounts.google.com/o/oauth2/auth',
+          token_uri: 'https://oauth2.googleapis.com/token',
+          auth_provider_x509_cert_url: 'https://www.googleapis.com/oauth2/v1/certs',
+          client_x509_cert_url: '',
+        }),
+      },
+      logger: loggingSystemMock.createLogger(),
+      services: mockServices,
+      request: mockKibanaRequest,
+    };
+
+    connector = new GeminiConnector(mockServiceParams);
+  });
+
+  describe('getDashboard', () => {
+    it('should return available: true when user has privileges', async () => {
+      const dashboardId = 'some-dashboard-id';
+
+      // Mock initDashboard to simulate a successful dashboard fetch
+      jest.spyOn(connector, 'getDashboard').mockResolvedValueOnce({ available: true });
+
+      const response = await connector.getDashboard({ dashboardId });
+      expect(response).toEqual({ available: true });
+    });
+
+    // Add a test for the case when the user doesn't have privileges
+  });
+
+  describe('runApi', () => {
+    it('should send a formatted request to the API and return the response', async () => {
+      const runActionParams: RunActionParams = {
+        body: JSON.stringify({
+          messages: [
+            {
+              contents: [
+                {
+                  role: 'user',
+                  parts: [{ text: 'What is the capital of France?' }],
+                },
+              ],
+            },
+          ],
+        }),
+        model: 'test-model',
+      };
+
+      const mockApiResponse: AxiosResponse<RunApiResponse> = {
+        data: {
+          candidates: [
+            {
+              content: {
+                parts: [{ text: 'Paris' }],
+              },
+            },
+          ],
+          usageMetadata: {
+            promptTokenCount: 4096,
+            candidatesTokenCount: 512,
+            totalTokenCount: 8192,
+          },
+        },
+        status: 200,
+        statusText: 'OK',
+        headers: {},
+        config: {
+          headers: mockAxiosHeaders,
+        },
+      };
+
+      // Mock the request function to simulate a successful API call
+      jest.spyOn(connector as GeminiConnector, 'request').mockResolvedValueOnce(mockApiResponse);
```

Comment:
```

x-pack/plugins/stack_connectors/server/connector_types/gemini/gemini.test.ts:217:48 - error TS2769: No overload matches this call.
--
  | proc [tsc]   Overload 1 of 3, '(object: GeminiConnector, method: never): SpyInstance<never, never, any>', gave the following error.
  | proc [tsc]     Argument of type 'string' is not assignable to parameter of type 'never'.
  | proc [tsc]   Overload 2 of 3, '(object: GeminiConnector, method: "getSubActions" \| "runApi" \| "getDashboard"): SpyInstance<Promise<Readonly<{} & { available: boolean; }>> \| Promise<...> \| Map<...>, [] \| ... 1 more ... \| [...], any>', gave the following error.
  | proc [tsc]     Argument of type '"request"' is not assignable to parameter of type '"getSubActions" \| "runApi" \| "getDashboard"'.
  | proc [tsc]
  | proc [tsc] 217       jest.spyOn(connector as GeminiConnector, 'request').mockResolvedValueOnce(mockApiResponse);
  | proc [tsc]                                                    ~~~~~~~~~
```

Check how we mock the request in `bedrock.test.ts`

Reply:
cc: @shivankawasthi11

Reply:
Fixed in the latest commit